analyzers/src/SonarAnalyzer.CSharp/Rules/UnusedPrivateMember.cs

```diff
@@ -215,7 +215,12 @@ variableDeclarator.Parent.Parent switch
                 };
 
             Diagnostic CreateS1144Diagnostic(SyntaxNode syntaxNode, ISymbol symbol) =>
-                Diagnostic.Create(RuleS1144, syntaxNode.GetLocation(), accessibility, symbol.GetClassification(), GetMemberName(symbol));
+                Diagnostic.Create(RuleS1144, GetIdentifierLocation(syntaxNode), accessibility, symbol.GetClassification(), GetMemberName(symbol));
+
+            static Location GetIdentifierLocation(SyntaxNode syntaxNode) =>
+                syntaxNode.GetIdentifier().HasValue
+                    ? syntaxNode.GetIdentifier().Value.GetLocation()
+                    : syntaxNode.GetLocation();
```

Comment:
```suggestion
                (syntaxNode.GetIdentifier() ?? syntaxNode).GetLocation();
```

Reply:
Unfortunately, it cannot be used like that. `GetIdentifier()` returns a `SyntaxToken`, while `syntaxNode` is of type `SyntaxNode`. They both - independently - define a `GetLocation()` method, but they are not the same method, even though they do exactly the same thing. Therefore the compilation fails.
Which cases do you refer, to where it might not work as intended?

Reply:
> Which cases do you refer, to where it might not work as intended?

[Local functions - C# Programming Guide | Microsoft Learn](https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/local-functions)

```cs
public void M()
{
   // ...
    void Unused() { } // Noncompliant
//       ^^^^^^       // <- "GetIdentifier" might miss this one
}
```

Reply:
> GetIdentifier() returns a SyntaxToken, while syntaxNode is of type SyntaxNode

Ah. I see. You should do this then:
```suggestion
                syntaxNode.GetIdentifier() is { } identifier
                    ? identifier.GetLocation()
                    : syntaxNode.GetLocation();
```

Reply:
That works! Refactored it.

Reply:
good catch!